app/code/Magento/CatalogRuleConfigurable/Plugin/CatalogRule/Model/ConfigurableProductsProvider.php

```diff
@@ -27,20 +27,22 @@ public function __construct(\Magento\Framework\App\ResourceConnection $resource)
     }
 
     /**
+     * Return list of ID for product variation
+     *
      * @param array $ids
      * @return array
      */
     public function getIds(array $ids)
     {
-        $key =  md5(json_encode($ids));
+        $key = md5(json_encode($ids)); //phpcs:ignore
```

Comment:
What are we ignoring here with `//phpcs:ignore`?

Reply:
Changing it to some other algorithm might cause some regression. It's not purpose of this PR of fixing using md5 there.